### DISCUSSION TITLE
Is there a way to separate text layer and image layer?

### DISCUSSION THREAD

**Discussion by amansani** [py-pdf/pypdf#1510] [Q&A] [answered]
- I am retrieving tables from a Original pdf file and each table is saved into another pdf file(for understanding we will call this as CroppedTableFromOriginal file) using mediabox cropping method. 
- When i am trying to get the table info from this CroppedTableFromOriginal file, the output consists of the text layer data too, which means it consists of other tables present in the page even after cropping. It's invisible in pdf, but can see the rest of other tables present in the page when dumping the data to csv/xlsx.
- I also tried other rectangle object methods, like crop box, bleed box etc.,. Nothing helped me.
- Is there a way we can separate the image layer and the text layer? I want to store only one cropped table info from this CroppedTableFromOriginal file to .xlsx or .csv formats avoiding rest of all other tables present in the background.
- Attached are the pdf files I am using. Any help is much appreciated. Thank you.

[Original.pdf](https://github.com/mstamy2/PyPDF2/files/4758943/Original.pdf)
[CroppedTableFromOriginal.pdf](https://github.com/mstamy2/PyPDF2/files/4758944/CroppedTableFromOriginal.pdf)

.

**Comment by amansani** [comment]
I got the solution by applying media box and crop box together. To fix this, we have to match boundaries of mediabox and cropbox. 

As it might help others in future, I am not deleting this issue. 

Thanks and hope it helps.

**Comment by lins575757** [comment]
> I got the solution by applying media box and crop box together. To fix this, we have to match boundaries of mediabox and cropbox.
> 
> As it might help others in future, I am not deleting this issue.
> 
> Thanks and hope it helps.

Hi, could you please explain in detail? I set mediaBox and cropBox at the same time, but I still can't solve it
<img width="445" alt="a" src="https://user-images.githubusercontent.com/26050860/106597234-17451a00-6591-11eb-9916-523bec7bd0e1.png">

**Comment by awa5114** [comment]
I also have the same issue, and applied the exact same modifications to `mediaBox`, `cropBox` and `trimBox`.

**Comment by jafa2222** [comment]
Were you able to solve this problem?
I have same issue



> > I got the solution by applying media box and crop box together. To fix this, we have to match boundaries of mediabox and cropbox.
> > As it might help others in future, I am not deleting this issue.
> > Thanks and hope it helps.
> 
> Hi, could you please explain in detail? I set mediaBox and cropBox at the same time, but I still can't solve it <img alt="a" width="445" src="https://user-images.githubusercontent.com/26050860/106597234-17451a00-6591-11eb-9916-523bec7bd0e1.png">

**Comment by pubpub-zz** [comment]
@jafa2222 
can you provide a full example : pdf source, pdf output (to ease analysis), standalone python  script in text format and what you think is wrong.

**Comment by jafa2222** [comment]
@pubpub-zz 
I want to extract different parts of Main file like the output file and save it in different sheets of Excel file. But when I crop the file, the text layer remains in all parts.

**Comment by pubpub-zz** [comment, marked as answer]
if you use the crop the text will still be present. The best is to use a visitor function with extract_text
see https://pypdf2.readthedocs.io/en/latest/user/extract-text.html#using-a-visitor

**Comment by jafa2222** [comment]
@pubpub-zz 
Thank you very much for your advice.
